packages/model-viewer/src/features/scene-graph/model.ts

```diff
@@ -226,16 +267,18 @@ export class Model implements ModelInterface {
     for (const primitive of this[$primitivesList]) {
       promises.push(primitive.enableVariant(variantName));
     }
-
     await Promise.all(promises);
 
-    for (const material of this.materials) {
-      material[$setActive](false);
-    }
-    // Marks the materials that are now in use after the variant switch.
-    for (const primitive of this[$primitivesList]) {
-      this.materials[primitive.getActiveMaterial().index][$setActive](true);
-    }
+    const updateActiveState = async () => {
```

Comment:
I  don't think this is doing what you think  it's doing. It's marked `async`, but it doesn't await anything, so it'll  return  immediately even if the  functions it calls are  also async.

Reply:
Ahh your right, forgot to revert this.